compiler/src/dmd/expressionsem.d

```diff
@@ -7300,6 +7300,17 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         {
             if (!hasOverloads && f.checkForwardRef(exp.loc))
                 return setError();
+            import dmd.builtin : determine_builtin;
+            if (!f.fbody)
+            {
+                const b = determine_builtin(f);
+                if (b != BUILTIN.unknown && b != BUILTIN.unimp)
+                {
+                    error(exp.loc, "cannot take address of intrinsic function `%s`", exp.e1.toChars());
+                    if (BUILTIN.sin <= b && b <= BUILTIN.isfinite)
+                        errorSupplemental(exp.loc, "use `&std.math.%s` instead", exp.e1.toChars());
+                }
```

Comment:
This is neither a comprehensive list, nor portable between different compilers.

Reply:
this is a problem indeed. The function used, `determine_builtin`, is based on certain module names, in theory it can return if it's a gcc or llvm intrindic, but it would be better to have a `@compilerProc` or `@intrinsic` attribute.

Reply:
@ibuclaw Would you accept this if `version(IN_GCC/LDC)` are treated properly or do you suggest a different course of action?

Reply:
I think a good starting point would be to put the logic that determines addressable-ness of an intrinsic into a separate function part of `dmd.builtins` - with C++ linkage so that gdc/ldc can override it.

As for the test, should be an intrinsic with no library symbol anywhere - alloca is a good example.

Reply:
I'm otherwise still unsure why an implementation detail should be a language error.